worker/provisioner/provisioner_test.go

```diff
@@ -1767,7 +1762,12 @@ func (s *ProvisionerSuite) TestAvailabilityZoneMachinesRestartTask(c *gc.C) {
 	defer stop(c, newTask)
 
 	availabilityZoneMachinesAfter := provisioner.GetAvailabilityZoneMachines(task)
-	c.Assert(availabilityZoneMachinesBefore, gc.DeepEquals, availabilityZoneMachinesAfter)
+	sort.Sort(provisioner.ByPopulationThenName(availabilityZoneMachinesBefore))
+	sort.Sort(provisioner.ByPopulationThenName(availabilityZoneMachinesAfter))
+	c.Assert(availabilityZoneMachinesBefore, gc.HasLen, len(availabilityZoneMachinesAfter))
+	for i, _ := range availabilityZoneMachinesBefore {
```

Comment:
rather than looping, can't you just do

`c.Assert(availabilityZoneMachinesBefore, jc.DeepEquals, availabilityZoneMachinesAfter)`

?

Reply:
jc.DeepEquals doesn't seem to work on a slice of pointers when you want to compare the contents of the pointers?  Or I just haven't figured part out yet.  I changed some other code around, made GetCopyAvailabilityZoneMachines() return a slice of structures instead of pointers , so that jc.DeepEquals would work.